src/OpenTelemetry.Contrib.Instrumentation.Wcf/WcfInstrumentationOptions.cs

```diff
@@ -0,0 +1,63 @@
+﻿// <copyright file="WcfInstrumentationOptions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.ServiceModel.Channels;
+using OpenTelemetry.Context.Propagation;
+
+namespace OpenTelemetry.Contrib.Instrumentation.Wcf
+{
+    /// <summary>
+    /// Options for WCF instrumentation.
+    /// </summary>
+    public class WcfInstrumentationOptions
+    {
+        /// <summary>
+        /// Gets or sets <see cref="TextMapPropagator"/> for context propagation. Default value: <see cref="CompositeTextMapPropagator"/> with <see cref="TraceContextPropagator"/> &amp; <see cref="BaggagePropagator"/>.
+        /// </summary>
+        public TextMapPropagator Propagator { get; set; } = new CompositeTextMapPropagator(new TextMapPropagator[]
+        {
+            new TraceContextPropagator(),
+            new BaggagePropagator(),
+        });
+
+        /// <summary>
+        /// Gets or sets a Filter function to filter instrumentation for requests on a per request basis.
+        /// The Filter gets the Message, and should return a boolean.
+        /// If Filter returns true, the request is collected.
+        /// If Filter returns false or throw exception, the request is filtered out.
+        /// </summary>
+        public Func<Message, bool> IncomingRequestFilter { get; set; }
```

Comment:
There's already a class System.ServiceModel.Dispatcher.MessageFilter with semantics to give a yes/no answer for filtering messages. WCF already has multiple public implementations such as an XPath based filter. Would it be worth using this instead? The downside is filters would need a class defined whereas this allows an anonymous delegate, but that could be easily solved by providing an implementation of MessageFilter which takes a delegate as a constructor argument to provide a low code solution.

Reply:
The filters on the options prevent spans (telemetry) from being created but they don't prevent messages from actually processing or anything like that. So I think it's a different thing from MessageFilter?

Reply:
I think suggestion here is to use `MessageFilter` class instead of a `Func<Message, bool>` as a more natural way to identify "interesting" messages

Reply:
@mconnew I'm not seeing MessageFilter exposed in the .NET Standard 2.0 surface should it be there?